tsdb/shard.go

```diff
@@ -580,6 +580,10 @@ func (s *Shard) validateSeriesAndFields(points []models.Point) ([]models.Point,
 				continue
 			}
 
+			// If the tags were created by models.ParsePointsWithPrecision,
+			// they refer to subslices of the buffer containing line protocol.
+			// To ensure we don't refer to that buffer, preventing that buffer from being garbage collected, clone the tags.
+			tags = tags.Clone()
```

Comment:
This is going to cause unnecessary  allocations a lot of the time because `CreateSeriesIndexIfNotExists` only holds onto the subsliced tags if a new series needs to be created.

I think it would be better to push `tags` through to `NewSeries` as-is, and then clone the tags inside `CreateSeriesIndexIfNotExists` only when `AddSeries` is called. Maybe you could just do something like:

```go
series.Tags = series.Tags.Clone()
m.AddSeries(series)
```